template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/atlas/AtlasKnoxRoleConfigProvider.java

```diff
@@ -38,11 +76,58 @@ public List<String> getRoleTypes() {
         return List.of("ATLAS_SERVER");
     }
 
-    @Override
-    public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
+    private boolean isAtlasKnoxConfigurationNeeded(TemplatePreparationObject source) {
         return Objects.nonNull(source.getGatewayView())
                 && Objects.nonNull(source.getGatewayView().getExposedServices())
                 && source.getGatewayView().getExposedServices().contains(exposedServiceCollector.getAtlasService().getKnoxService());
     }
 
+    private boolean isSDXOptimizationEnabled(TemplatePreparationObject source) {
+        return source.getStackType().equals(StackType.DATALAKE)
+                && entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId())
+                && isDatalakeVersionSupported(source.getBlueprintView().getBlueprintText());
```

Comment:
This change is not shape dependent.

Reply:
This change does not check, the shape. Just the version.
It's can be micro/light/medium duty. 

The stack should be `DATALAKE`. Because the ticket only mentioned the `Datalakes`